core/test/audits/has-hsts-test.js

```diff
@@ -0,0 +1,447 @@
+  const artifacts = {
+    devtoolsLogs: {
+      defaultPass: networkRecordsToDevtoolsLog([
+        {
+          url: 'https://example.com',
+          responseHeaders: [
+            {name: 'Strict-Transport-Security', value: `includeSubDomains; preload`},
+          ],
+        },
+      ]),
+    },
+    URL: {
+      requestedUrl: 'https://example.com',
+      mainDocumentUrl: 'https://example.com',
+      finalDisplayedUrl: 'https://example.com',
+    },
+  };
+
+  const results = await HasHsts.audit(artifacts, {computedCache: new Map()});
+  expect(results.notApplicable).toBeFalsy();
+  expect(results.details.items).toMatchObject([
+    {
+      severity: {
+        i18nId: "core/lib/i18n/i18n.js | itemSeverityHigh",
+        values: undefined,
+        formattedDefault: 'High'
+      },
+      description: {
+        i18nId: "core/audits/has-hsts.js | noMaxAge",
+        values: undefined,
+        formattedDefault: 'No max-age directive'
+      },
+      directive: 'max-age',
+    },
+  ]);
+});
+
+it('max-age too low, but other directives present', async () => {
+  const artifacts = {
+    devtoolsLogs: {
+      defaultPass: networkRecordsToDevtoolsLog([
+        {
+          url: 'https://example.com',
+          responseHeaders: [
+            {name: 'Strict-Transport-Security', value: `max-age=1337; includeSubDomains; preload`},
+          ],
+        },
+      ]),
+    },
+    URL: {
+      requestedUrl: 'https://example.com',
+      mainDocumentUrl: 'https://example.com',
+      finalDisplayedUrl: 'https://example.com',
+    },
+  };
+
+  const results = await HasHsts.audit(artifacts, {computedCache: new Map()});
+  expect(results.notApplicable).toBeFalsy();
+  expect(results.details.items).toMatchObject([
+    {
+      severity: {
+        i18nId: "core/lib/i18n/i18n.js | itemSeverityHigh",
+        values: undefined,
+        formattedDefault: 'High'
+      },
+      description: {
+        i18nId: "core/audits/has-hsts.js | lowMaxAge",
+        values: undefined,
+        formattedDefault: 'Max-age too low'
+      },
+      directive: 'max-age',
+    },
+  ]);
+});
+
+it('includeSubDomains missing, but other directives present', async () => {
+  const artifacts = {
+    devtoolsLogs: {
+      defaultPass: networkRecordsToDevtoolsLog([
+        {
+          url: 'https://example.com',
+          responseHeaders: [
+            {name: 'Strict-Transport-Security', value: `max-age=63072000; preload`},
+          ],
+        },
+      ]),
+    },
+    URL: {
+      requestedUrl: 'https://example.com',
+      mainDocumentUrl: 'https://example.com',
+      finalDisplayedUrl: 'https://example.com',
+    },
+  };
+
+  const results = await HasHsts.audit(artifacts, {computedCache: new Map()});
+  expect(results.notApplicable).toBeFalsy();
+  expect(results.details.items).toMatchObject([
+    {
+      severity: {
+        i18nId: "core/lib/i18n/i18n.js | itemSeverityMedium",
+        values: undefined,
+        formattedDefault: 'Medium'
```

Comment:
Pro tip: we have a custom assertion check for display strings ([example](https://github.com/GoogleChrome/lighthouse/blob/main/core/test/audits/bf-cache-test.js#L40))

So this could be `expect(results.details.items[0].severity).toBeDisplayString('Medium')`. Then you can drop the `i18nId`/`values` stuff from expectations.

Reply:
Thanks for the recommendation! I think I covered it across the whole test file.